admin/src/js/controllers/edit-user.js

```diff
@@ -254,24 +279,63 @@ angular
       return hasPlace && hasContact;
     };
 
+    const validateFacilityHierarchy = () => {
+      const placeIds = $scope.editUserModel.place;
+
+      if (!placeIds || placeIds.length === 1) {
+        return $q.resolve(true);
+      }
+
+      const getPlace = (placeId) => {
+        return DB().get(placeId);
+      };
+
+      return Promise.all(placeIds.map(getPlace))
+        .then((places) => {
+          const contactTypes = places.map(place => place.contact_type);
+          const isSameHeirarchy = contactTypes.every(type => type === contactTypes[0]);
```

Comment:
Um. This is pretty hard to understand. 
I think it would be easier to do something like this: 
```
const contactTypes = new Set(places.map(place => place.contact_type));
const isSameHeirarchy = contactTypes.size === 1;
```